deltachat-ios/Controller/ChatListController.swift

```diff
@@ -445,13 +445,15 @@ class ChatListController: UITableViewController {
         let archiveActionTitle: String = String.localized(archived ? "unarchive" : "archive")
 
         let archiveAction = UITableViewRowAction(style: .destructive, title: archiveActionTitle) { [weak self] _, _ in
-            self?.viewModel?.archiveChatToggle(chatId: chatId)
+            self?.viewModel?.archiveChatToggle(chatId: chatId, notifyListener: true)
+            self?.setEditing(false, animated: true)
```

Comment:
i do not really get why `setEditing(false)` needs to be called here - this "edit" mode refers to the new multi-select seems unrelated to the "old swipe" action.

is it for a quick redraw?

Reply:
exactly, it is for a quick redraw. otherwise setEditing is set to false by the viewcontroller itself (internally) once the swipe back animation has finished.

Reply:
so, the "edit mode" is not only for multi-select but also for the swipe actions?

i am still wondering why before multi-select the swipe-to-pin and swipe-to-archive was faster even with the "closing animation".

Reply:
because always all list updates have been done immediately. With multi-select we postpone list-updates until editing is done. This way we avoid that the selected items change (we use the tableViewController own mecahnism to track the selected cells - not the selected messages).